File: cpp/day1/problem2/solve.cpp
#include <iostream>
#include <fstream>
#include <vector>

int main(void)
{
	std::ifstream myFile("../numbers.txt");
	std::string myString;

	if (!myFile.is_open()) {
		std::cout << "Unable to open file\n";
		return 1;
	}

	std::vector<int> depths = {};
	while (getline(myFile, myString)) {
		depths.push_back(std::stoi(myString));		
	}

	int increases = 0;
	for (int i = 3; i < depths.size(); i++) {
		if (depths[i] > depths[i - 3]) {
			increases++;
		}
	}

	std::cout << increases << "\n";

	myFile.close();

	return 0;
}

File: go/day1/problem2/solve.go
package main

import (
	"log"
	"fmt"
	"bufio"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("../numbers.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	count := 0
	lines := make([]int, 0)

	for scanner.Scan() {
		num, err := strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println(err)
			os.Exit(2)
		}

		lines = append(lines, num)
	}

	for i := 3; i < len(lines); i++ {
		if lines[i] > lines[i - 3] {
			count++
		}
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	fmt.Printf("Increases: %d\n", count)
}
